src/stacked_borrows.rs

```diff
@@ -466,7 +444,8 @@ impl<'tcx> Stack {
         })?;
 
         // Step 2: Remove all items.  Also checks for protectors.
```

Comment:
This doesn't "remove" any more.

I was thinking of this as being similar to the popping on access... but since the memory goes away I guess it is okay to implement it very differently.

Reply:
What do you mean? I'm pretty sure the caching system honors removal properly, or are you referring to something with Wildcard?

Reply:
What I meant is they don't actually remove things from the stack, they just iterate over the entire stack and do something for each element.